From 26d428b924aa1445604fbeff2f0509b1519ee4b0 Mon Sep 17 00:00:00 2001
From: Navarum Eliskar <48705414+navarum.noreply.github.com>
Date: Thu, 21 Mar 2019 00:12:31 -0700
Subject: [PATCH 5/7] Patchlevel

---
 src/patchlevel.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/patchlevel.h b/src/patchlevel.h
index 5ce1300..0912d9e 100644
--- a/src/patchlevel.h
+++ b/src/patchlevel.h
@@ -525,9 +525,9 @@
  * 2005-12-19, 4.00.03jw3 syntax error.
  */
 
-#define ORIGIN "FAU"
+#define ORIGIN ""
 #define REV 4
 #define VERS 0
 #define PATCHLEVEL 3
-#define DATE "2-May-06"
-#define STATE "jw4"
+#define DATE "20-Mar-19"
+#define STATE "ne"
-- 
2.21.0

