From 8f131c4edf848b1ca2da81e3d03f07f84c48bbe6 Mon Sep 17 00:00:00 2001
From: Navarum <48705414+navarum@users.noreply.github.com>
Date: Wed, 9 Oct 2019 13:08:23 -0700
Subject: [PATCH 2/4] Make a note of configure flags

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 0da3b596..bd6fa168 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7,6 +7,11 @@ AC_PREREQ(2.50)
 AC_CONFIG_SRCDIR([src/feature.h])
 AC_CONFIG_HEADER(config.h:config.h.in)
 
+# NE 09 Oct 2019
+# https://unix.stackexchange.com/questions/35905/autoconf-save-configure-command-line-to-config-h
+config_flags="$*"
+AC_DEFINE_UNQUOTED([CONFIG_FLAGS],["$config_flags"],[Flags passed to configure])
+
 AC_CANONICAL_HOST
 
 dnl RXVT version
-- 
2.23.0

